akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -532,7 +532,9 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         if (shell.isInitialized) {
           // yes, this steals another shell’s Resume, but that’s okay because extra ones will just not do anything
           finishShellRegistration()
-        } else tryInit(shell)
+        } else if (!tryInit(shell)) {
```

Comment:
Yes: I hate hiding side-effects among simple tests. Sometimes it is hard to avoid, but here I think this is a lot clearer.